GVFS/GVFS.FunctionalTests.LockHolder/AcquireGVFSLock.cs

```diff
@@ -45,6 +41,43 @@ public void Execute()
             }
         }
 
+        private static bool TryGetGVFSEnlistmentRootImplementation(string directory, out string enlistmentRoot, out string errorMessage)
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                return MacPlatform.TryGetGVFSEnlistmentRootImplementation(directory, out enlistmentRoot, out errorMessage);
+            }
+
+            // Not able to use WindowsPlatform here - because of its dependency on WindowsIdentity (and also kernel32.dll).
+            enlistmentRoot = null;
+
+            string finalDirectory;
+            if (!WindowsFileSystem.TryGetNormalizedPathImplementation(directory, out finalDirectory, out errorMessage))
+            {
+                return false;
+            }
+
+            enlistmentRoot = Paths.GetRoot(finalDirectory, GVFSConstants.DotGVFS.Root);
+            if (enlistmentRoot == null)
+            {
+                errorMessage = $"Failed to find the root directory for {GVFSConstants.DotGVFS.Root} in {finalDirectory}";
+                return false;
+            }
+
+            return true;
+        }
+
+        private static string GetNamedPipeNameImplementation(string enlistmentRoot)
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                return MacPlatform.GetNamedPipeNameImplementation(enlistmentRoot);
+            }
+
+            // Not able to use WindowsPlatform here - because of its dependency on WindowsIdentity (and also kernel32.dll).
+            return "GVFS_" + enlistmentRoot.ToUpper().Replace(':', '_');

```

Comment:
There is now code that we have to keep in sync.  Add a comment here pointing to the code in `WindowsPlatform`  and a comment in `WindowsPlatform` pointing to this code so that if we ever change one we will know to change the other.

Reply:
Agree. We discussed this - please see @wilbaker's comments above. 

(With how limited this is though I think you'd be OK just adding a comment here that describes why the code was copied rather than calling WindowsPlatform. As general rule the functional test code doesn't consume product code anyway, this project is the one exception.

Once everything moves to .NET Core for the WindowsIdentity issue we can look at the compatibility pack described here: https://devblogs.microsoft.com/dotnet/announcing-the-windows-compatibility-pack-for-net-core/)